plan/stats.go

```diff
@@ -254,9 +271,59 @@ func (p *LogicalProjection) deriveStats() (*statsInfo, error) {
 		cols := expression.ExtractColumns(expr)
 		p.stats.cardinality[i] = getCardinality(cols, p.children[0].Schema(), childProfile)
 	}
+	// If we enables the enhance selectivity we'll try to maintain the histogram.
+	if p.ctx.GetSessionVars().OptimizerSelectivityLevel >= 1 && !childProfile.histColl.Pseudo {
```

Comment:
should we check `childProfile.histColl != nil` firstly?

Reply:
Ok, it should. Maybe set its default value as a `PseudoHistogram` can save the code.